### PR TITLE
Added .Auto to ImageFormat to enhance GetAvatarUrl

### PR DESCRIPTION
**Pull request by KoB-Kirito**
Make sure you familiarize yourself with our contributing guidelines.

# Summary
Adds .Auto to ImageFormat to enhance the use of GetAvatarUrl

# Details
There is no convenient way to get the avatar url of an user with a specific size without checking if it's animated or not. With this change you can use GetAvatarUrl for that.

# Changes proposed
See diff. No breaking changes.

# Notes
It's quite simple~

### CODE REVIEW

File: DSharpPlus/Entities/DiscordUser.cs [hunk 1]
```diff
@@ -194,6 +194,9 @@ public string GetAvatarUrl(ImageFormat fmt, ushort size = 1024)
                     sfmt = "webp";
                     break;
 
+                case ImageFormat.Auto:
+                    sfmt = !string.IsNullOrWhiteSpace(this.AvatarHash) ? (AvatarHash.StartsWith("a_") ? "gif" : "png") : "png";
```
Comment (Kiritsu) [inline DSharpPlus/Entities/DiscordUser.cs:198]: Be consistent and add `this` the the second `AvatarHash`

File: DSharpPlus/Entities/DiscordUser.cs [hunk 2]
```diff
@@ -194,6 +194,9 @@ public string GetAvatarUrl(ImageFormat fmt, ushort size = 1024)
...
+                    break;
                 default:
```
Comment (Kiritsu) [inline DSharpPlus/Entities/DiscordUser.cs:200]: Tiny nitpick but you could add a line break like after every other case.

Comment (Emzi0767) [reply DSharpPlus/Entities/DiscordUser.cs:200]: You should actually, per style guidelines

File: DSharpPlus/Entities/DiscordUser.cs (hunk 1 above)
Comment (KoB-Kirito) [reply DSharpPlus/Entities/DiscordUser.cs:198]: It was like this for AvatarUrl too. Thought that has some kind of purpose. Should I change it then there too?

**Comment by Emzi0767** [comment]
Also I should add that there's several places where this enum is used. You should ensure the new value is appropriately handled in all places it's used.

**Comment by KoB-Kirito** [comment]
I checked every location it's used and handled it there accordingly.

**Comment by Emzi0767** [comment]
I'll do a deeper look later.

**Pull request merged**